core/sched.c

```diff
@@ -221,3 +229,21 @@ NORETURN void sched_task_exit(void)
     sched_active_thread = NULL;
     cpu_switch_context_exit();
 }
+
+#ifdef MODULE_CORE_PRIORITY_INHERITANCE
+void sched_change_priority(thread_t *thread, uint8_t priority)
+{
+    assert(thread && (priority < SCHED_PRIO_LEVELS));
```

Comment:
I suggest to avoid using an `assert` here (I'm wondering about performance). Actually, I suggest to remove any kind of check here. The function shall be used with care (which is already the case :-))